src/expressions/match-expr.md

```diff
@@ -96,6 +96,27 @@ Every binding in each `|` separated pattern must appear in all of the patterns i
 r[expr.match.binding-restriction]
 Every binding of the same name must have the same type, and have the same binding mode.
 
+r[expr.match.type]
+The type of the overall `match` expression is the [least upper bound](../type-coercions.md#r-coerce.least-upper-bound) of the individual match arms.
+
+r[expr.match.empty]
+If there are no match arms, then the `match` expression is diverging and the type is [`!`](../types/never.md).
```

Comment:
does this require a diverging scrutinee expression to compile?

probably worth adding an example either way

Reply:
I think it's useful to mention this, thoughts on this: https://github.com/rust-lang/reference/pull/2067/commits/fa9825d290f8a58ff7850a339f1955ff7fa14db2

Reply:
Changed this to be a `NOTE`.